In [21]:
#Importing libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

## Answer-1-The population of each state/ut

#### This data has been collected from http://cyberjournalist.org.in/census/cenindia.html for year 2001 on population

In [22]:
dp=pd.read_csv(r"C:\Users\OM RAJ PANDEY\Documents\Population data-2001.csv")
dp.head(37)

Unnamed: 0                       State   Population       Male   \
0          1.0           Jammu and Kashmir      10069917    5300574   
1          2.0            Himachal Pradesh       6077248    3085256   
2          3.0                      Punjab      24289296   12963362   
3          4.0                  Chandigarh        900914     508224   
4          5.0                 Uttaranchal       8479562    4316401   
5          6.0                     Haryana      21082989   11327658   
6          7.0                       Delhi      13782976    7570890   
7          8.0                   Rajasthan      56473122   29381657   
8          9.0               Uttar Pradesh     166052859   87466301   
9         10.0                       Bihar      82878796   43153964   
10        11.0                      Sikkim        540493     288217   
11        12.0           Arunachal Pradesh       1091117     573951   
12        13.0                    Nagaland       1988636    1041686   
13        14.0                     Manipur       2388634    1207338   
14        15.0                     Mizoram        891058     459783   
15        16.0                     Tripura       3191168    1636138   
16        17.0                   Meghalaya       2306069    1167840   
17        18.0                       Assam      26638407   13787799   
18        19.0                 West Bengal      80221171   41487694   
19        20.0                   Jharkhand      26909428   13861277   
20        21.0                      Orissa      36706920   18612340   
21        22.0                 Chhatisgarh      20795956   10452426   
22        23.0              Madhya Pradesh      60385118   31456873   
23        24.0                     Gujarat      50596992   26344053   
24        25.0                 Daman & Diu        158059      92478   
25        26.0        Dadra & Nagar Haveli        220451     121731   
26        27.0                 Maharashtra      96752247   50334270   
27        28.0              Andhra Pradesh      75727541   38286811   
28        29.0                   Karnataka      52733958   26856343   
29        30.0                         Goa       1343998     685617   
30        31.0                 Lakshadweep         60595      31118   
31        32.0                      Kerala      31838619   15468664   
32        33.0                  Tamil Nadu      62110839   31268654   
33        34.0                 Pondicherry        973829     486705   
34        35.0  Andaman & Nicobar Islands*        356265     192985   
35         NaN                INDIA Total     1027015247  531277078   

       Female  Urban  
0     4769343  24.85  
1     2991992   9.79  
2    11325934  33.95  
3      392690  89.78  
4     4163161  25.59  
5     9755331  29.00  
6     6212086  93.01  
7    27091465  23.38  
8    78586558  20.78  
9    39724832  10.47  
10     252276  11.10  
11     517166  20.41  
12     946950  17.74  
13    1181296  23.88  
14     431275  49.50  
15    1555030  17.02  
16    1138229  19.63  
17   12850608  12.72  
18   38733477  28.03  
19   13048151  22.25  
20   18094580  14.97  
21   10343530  20.08  
22   28928245  26.67  
23   24252939  37.35  
24      65581  36.26  
25      98720  22.89  
26   46417977  42.40  
27   37440730  27.08  
28   25877615  33.98  
29     658381  49.47  
30      29477  44.47  
31   16369955  25.97  
32   30842185  43.86  
33     487124  66.57  
34     163280  32.67  
35  495738169  27.78

#### This data has been collected from https://censusindia.gov.in/census.website/data/census-tables for year 2011.So,for census
#### counting has been done in the gap of every 10 years.So this data is for 2001 to 2011.

#### THE POUPLATION OF EACH STATE-2011

In [36]:
df=pd.read_csv(r"C:\Users\OM RAJ PANDEY\Desktop\POPULATION DATA-2011.csv")
df.head(37)


COUNTRY/STATE/UT  YEAR    TOTAL PERSON          MALE        FEMALE
0                  INDIA  2011  1,21,08,54,977  62,32,70,258  58,75,84,719
1       HIMACHAL PRADESH  2011     1,25,41,302     66,40,662     59,00,640
2                 PUNJAB  2011     2,77,43,338   1,46,39,465   1,31,03,873
3             CHANDIGARH  2011       10,55,450      5,80,663      4,74,787
4            UTTARAKHAND  2011     1,00,86,292     51,37,773     49,48,519
5                HARYANA  2011     2,53,51,462   1,34,94,734   1,18,56,728
6           NCT OF DELHI  2011     1,67,87,941     89,87,326     78,00,615
7              RAJASTHAN  2011     6,85,48,437   3,55,50,997   3,29,97,440
8          UTTAR PRADESH  2011    19,98,12,341  10,44,80,510   9,53,31,831
9                  BIHAR  2011    10,40,99,452   5,42,78,157   4,98,21,295
10                SIKKIM  2011        6,10,577      3,23,070      2,87,507
11     ARUNACHAL PRADESH  2011       13,83,727      7,13,912      6,69,815
12              NAGALAND  2011       19,78,502     10,24,649      9,53,853
13               MANIPUR  2011       28,55,794     14,38,586     14,17,208
14               MIZORAM  2011       10,97,206      5,55,339      5,41,867
15               TRIPURA  2011       36,73,917     18,74,376     17,99,541
16             MEGHALAYA  2011       29,66,889     14,91,832     14,75,057
17                 ASSAM  2011     3,12,05,576   1,59,39,443   1,52,66,133
18           WEST BENGAL  2011     9,12,76,115   4,68,09,027   4,44,67,088
19             JHARKHAND  2011     3,29,88,134   1,69,30,315   1,60,57,819
20                ODISHA  2011     4,19,74,218   2,12,12,136   2,07,62,082
21           CHHATISGARH  2011     2,55,45,198   1,28,32,895   1,27,12,303
22        MADHYA PRADESH  2011     7,26,26,809   3,76,12,306   3,50,14,503
23                GUJRAT  2011     6,04,39,692   3,14,91,260   2,89,48,432
24           DAMAN & DIU  2011        2,43,247      1,50,301        92,946
25  DADAR & NAGAR HAVELI  2011        3,43,709      1,93,760      1,49,949
26           MAHARASHTRA  2011    11,23,74,333   5,82,43,056   5,41,31,277
27        ANDHRA PRADESH  2011     8,45,80,777   4,24,42,146   4,21,38,631
28              KARNATKA  2011     6,10,95,297   3,09,66,657   3,01,28,640
29                   GOA  2011       14,58,545      7,39,140      7,19,405
30           LAKSHADWEEP  2011          64,473        33,123        31,350
31                KERALA  2011     3,34,06,061   1,60,27,412   1,73,78,649
32             TAMILNADU  2011     7,21,47,030   3,61,37,975   3,60,09,055
33            PODUCHERRY  2011       12,47,953      6,12,511      6,35,442
34     ANDAMAN & NICOBAR  2011        3,80,581      2,02,871      1,77,710
35       JAMMU & KASHMIR  2011     1,25,41,302     66,40,662     59,00,640

## ANSWER-2- The literacy rate of each state/ut

#### This data has been collected from https://socialjustice.gov.in/common/76669 for year 2001 on literacy rate of India

In [17]:
dl=pd.read_csv(r"C:\Users\OM RAJ PANDEY\Documents\Literacy rate-2001.csv")
dl.head(37)

State/UT Literacy Rate 2001 Unnamed: 2 Unnamed: 3
0                    NaN              Total         SC         PD
1       1. Uttar Pradesh               56.3       46.3       44.5
2         2. Maharashtra               76.9       71.9       57.6
3               3. Bihar                 47       28.5       37.2
4         4. West Bengal               68.6         59       52.7
5      5. Andhra Pradesh               60.5       53.5       44.2
6          6. Tamil Nadu               73.5       63.2       58.6
7      7. Madhya Pradesh               63.7       58.6       49.7
8           8. Rajasthan               60.1       52.2       45.3
9           9. Karnataka               66.6       52.9       50.3
10           10. Gujarat               69.1       70.5         54
11            11. Orissa               63.1       55.5       48.8
12            12. Kerala               90.9       82.7         67
13         13. Jharkhand               53.6       37.6       41.5
14             14. Assam               63.3       66.8       45.5
15            15. Punjab               69.7       56.2       49.5
16           16. Haryana               57.9       55.4       48.8
17      17. Chhattisgarh               64.7         64       47.9
18             18. Delhi               81.7       70.8       65.5
19               19. J&K               55.5         59       39.9
20       20. Uttarakhand               71.6       63.4       51.8
21  21. Himachal Pradesh               76.5       70.3       51.9
22           22. Tripura               73.2       74.7       52.5
23           23. Manipur               70.5       72.3       42.9
24         24. Meghalaya               62.5       56.3       37.9
25          25. Nagaland               66.6        NaN       30.8
26               26. Goa                 82       71.9         50
27      27. Aru .Pradesh               54.3       67.6       39.4
28        28. Puducherry               81.2       69.1       57.7
29           29. Mizoram               88.8       89.2       56.3
30        30. Chandigarh               81.9       67.7       62.5
31            31. Sikkim               68.8         63         40
32     32. A & N Islands               81.3        NaN       57.1
33  33. D & Nagar Haveli               57.6       78.2         25
34       34. Daman & Diu               78.2       85.1       33.3
35       35. Lakshadweep               86.7        NaN         45
36                 Total               64.8       54.7       49.3

#### LITERACY RATE IN EACH STATE-2011

#### This data has been collected from https://www.census2011.co.in/literacy.php for year 2011

In [14]:
df=pd.read_csv(r"C:\Users\OM RAJ PANDEY\Documents\Literacy rate.csv")
df.head(37)

Index             Country/State/ut  Literacy   Male  Female  % Change
0       0                        India     74.04  82.14   65.46      8.66
1       1                       Kerala     94.00  96.11   92.07      3.14
2       2                  Lakshadweep     91.85  95.56   87.95      5.19
3       3                      Mizoram     91.33  93.35   89.27      2.53
4       4                          Goa     88.70  92.65   84.66      6.69
5       5                      Tripura     87.22  91.53   82.73     14.03
6       6                Daman and Diu     87.10  91.54   79.55      8.92
7       7  Andaman and Nicobar Islands     86.63  90.27   82.43      5.33
8       8                        Delhi     86.21  90.94   80.76      4.54
9       9                   Chandigarh     86.05  89.99   81.19      4.11
10     10                   Puducherry     85.85  91.26   80.67      4.61
11     11             Himachal Pradesh     82.80  89.53   75.93      6.32
12     12                  Maharashtra     82.34  88.38   75.87      5.46
13     13                       Sikkim     81.42  86.55   75.61     12.61
14     14                   Tamil Nadu     80.09  86.77   73.44      6.64
15     15                     Nagaland     79.55  82.75   76.11     12.96
16     16                  Uttarakhand     78.82  87.40   70.01      7.20
17     17                      Gujarat     78.03  85.75   69.68      8.89
18     18                      Manipur     76.94  83.58   70.26     10.33
19     19                  West Bengal     76.26  81.69   70.54      7.62
20     20       Dadra and Nagar Haveli     76.24  85.17   64.32     18.61
21     21                       Punjab     75.84  80.44   70.73      6.19
22     22                      Haryana     75.55  84.06   65.94      7.64
23     23                    Karnataka     75.36  82.47   68.08      8.72
24     24                    Meghalaya     74.43  75.95   72.89     11.87
25     25                       Orissa     72.87  81.59   64.01      9.79
26     26                        Assam     72.19  77.85   66.27      8.94
27     27                 Chhattisgarh     70.28  80.27   60.24      5.62
28     28               Madhya Pradesh     69.32  78.73   59.24      5.58
29     29                Uttar Pradesh     67.68  77.28   57.18     11.41
30     30            Jammu and Kashmir     67.16  76.75   56.43     11.64
31     31               Andhra Pradesh     67.02  74.88   59.15      6.55
32     32                    Jharkhand     66.41  76.84   55.42     12.85
33     33                    Rajasthan     66.11  79.19   52.12      5.70
34     34            Arunachal Pradesh     65.38  72.55   57.70     11.04
35     35                        Bihar     61.80  71.20   51.50     14.80

#### ANSWER NO.3-THE AREA OF EACH STATE/UT

#### This data has been collected form https://censusindia.gov.in/census.website/data/census-tables for year 2001

In [18]:
da=pd.read_excel(r"C:\Users\OM RAJ PANDEY\Desktop\PC01_A01_Statement1.xls")
da.head(40)

Sl. No.      State/Union Territory #     Area in sq. km.      Unnamed: 3  \
0       NaN                          NaN               Total           Rural   
1       1.0                            2                   3               4   
2       NaN                          NaN                 NaN             NaN   
3       NaN                 INDIA  @ + $             3287263  3,118,237.74 *   
4       NaN                          NaN                 NaN             NaN   
5       1.0            Jammu & Kashmir @              222236       221272.69   
6       2.0             Himachal Pradesh               55673        55431.29   
7       3.0                       Punjab               50362        48265.38   
8       4.0                 Chandigarh #                 114           34.66   
9       5.0                  Uttaranchal               53483        52686.23   
10      6.0                      Haryana               44212        42931.63   
11      7.0                N.C.T. Delhi                 1483          558.32   
12      8.0                    Rajasthan              342239       336807.81   
13      9.0                Uttar Pradesh              240928       234369.81   
14     10.0                        Bihar               94163         92358.4   
15     11.0                       Sikkim                7096            N.A.   
16     12.0            Arunachal Pradesh               83743            N.A.   
17     13.0                     Nagaland               16579        16431.76   
18     14.0                      Manipur               22327        22176.81   
19     15.0                      Mizoram               21081        20494.08   
20     16.0                      Tripura               10486         10345.6   
21     17.0                    Meghalaya               22429        22199.28   
22     18.0                        Assam               78438        77476.23   
23     19.0                  West Bengal               88752        85427.26   
24     20.0                    Jharkhand               79714        77921.57   
25     21.0                       Orissa              155707       152912.87   
26     22.0                 Chhattisgarh              135191       133325.25   
27     23.0               Madhya Pradesh              308245       301283.24   
28     24.0                      Gujarat              196024        190796.5   
29     25.0                Daman & Diu #                 112           88.64   
30     26.0       Dadra & Nagar Haveli #                 491          473.83   
31     27.0                  Maharashtra              307713        300357.2   
32     28.0               Andhra Pradesh              275045        270297.5   
33     29.0                    Karnataka              191791       186590.02   
34     30.0                          Goa                3702         3190.33   
35     31.0                Lakshadweep #                  32           21.41   
36     32.0                       Kerala               38863        35611.14   
37     33.0                   Tamil Nadu              130058       117532.75   
38     34.0                Pondicherry #                 479          345.59   
39     35.0  Andaman & Nicobar Islands #                8249         8222.66   

          Unnamed: 4       Proportion to                Rank  
0              Urban  total area of India                NaN  
1                  5                    6                  7  
2                NaN                  NaN                NaN  
3        78,163.26 *                  100               ----  
4                NaN                  NaN                NaN  
5             963.31                 6.76                  6  
6             241.71                 1.69                 17  
7            2096.62                 1.53                 19  
8              79.34                 N.A.                 33  
9             796.77                 1.63                 18  
10           1280.37          

#### This data has been collected from https://www.indiastat.com/specimen-tables/geographical-data for year 2011

In [15]:
da=pd.read_csv(r"C:\Users\OM RAJ PANDEY\Documents\Area.csv")
da.head(37)

States/UTs     Total    Rural   Urban
0   Andaman and Nicobar Islands     8249     8211      38
1                Andhra Pradesh   275045   267191    7855
2             Arunachal Pradesh    83743     N.A.    N.A.
3                         Assam    78438    77178    1260
4                         Bihar    94163    91838    2325
5                    Chandigarh      114        4     110
6                  Chhattisgarh   135192   131810    3382
7        Dadra and Nagar Haveli      491      445      46
8                 Daman and Diu      111       56      55
9                         Delhi     1483      326    1157
10                          Goa     3702     2903     799
11                      Gujarat   196244   188840    7404
12                      Haryana    44212    42236    1976
13             Himachal Pradesh    55673    55402     271
14            Jammu and Kashmir   222236   220990    1246
15                    Jharkhand    79716    77467    2249
16                    Karnataka   191791   185783    6008
17                       Kerala    38852    31253    7599
18                  Lakshadweep       30        8      22
19               Madhya Pradesh   308252   300506    7746
20                  Maharashtra   307713   298629    9084
21                      Manipur    22327    22148     180
22                    Meghalaya    22429    22146     283
23                      Mizoram    21081    20494     587
24                     Nagaland    16579    16336     243
25                       Odisha   155707   152355    3352
26                   Puducherry      490      335     155
27                       Punjab    50362    47847    2515
28                    Rajasthan   342239   335606    6633
29                       Sikkim     7096     7058      38
30                   Tamil Nadu   130060   116428   13632
31                      Tripura    10486    10094     392
32                Uttar Pradesh   240928   233366    7562
33                  Uttarakhand    53483    52581     902
34                  West Bengal    88752    83633    5119
35                        India  3287469  3101505  102221

# OBSERVATIONS:According to 2001 and 2011 census data.
### 1. Jammu & Kashmir population was increased by 24,71,385 ,  literacy rate was increased by 11.6% and area is same.
### 2. Himachal pradesh population was increased by 64,64,054 ,  literacy rate was increased by 6.3% and area is same. 
### 3. Punjab population was increased by 34,54,042 ,  literacy rate was increased by 6.14% and area is same.
### 4. Chandigarh population was increased by 1,54,536,  literacy rate was increased by 4.15% and area is same.
### 5. Bihar population was increased by 21,22,065,  literacy rate was increased by 14.8% and area is same.
### 6. Uttar pradesh population was increased by 3,37,59,482,  literacy rate was increased by 11.38% and area is same.
### 7. Rajasthan population was increased by 1,20,75,315,  literacy rate was increased by 6.01% and area is same.
### 8. Delhi population was increased by 30,04,965,  literacy rate was increased by 4.51% and area is same.
### 9. Haryana population was increased by 42,68,473,  literacy rate was increased by 17.65% and area is same.
### 10. Uttarakhand population was increased by 16,06,730,  literacy rate was increased by 7.22% and area is same.
### 11. Meghalaya population was increased by 6,60,820,  literacy rate was increased by 11.93% and area is same.
### 12. Tripura population was increased by 4,82,749,  literacy rate was increased by 14.02% and area is same.
### 13. Mizoram population was increased by 2,06,148,  literacy rate was increased by 2.53% and area is same.
### 14. Manipur population was increased by 4,67,160,  literacy rate was increased by 6.44% and area is same.
### 15. Nagaland population was *decreased by 10,134,  literacy rate was increased by 12.95% and area is same.
### 16. Arunanchal pradesh population was increased by 1,27,46,610,  literacy rate was increased by 11.08% and area is same.
### 17. Sikkim population was increased by 70,084,  literacy rate was increased by 12.62% and area is same.
### 18. Madhya pradesh population was increased by 1,22,41,691,  literacy rate was increased by 5.62% and area is same.
### 19. Chhatisgarh population was increased by 47,49,242,  literacy rate was increased by 5.58% and area is increased by 1 sq.km.
### 20. Orissa population was increased by 52,67,298,  literacy rate was increased by 9.77% and area is same.
### 21. Jharkhand population was increased by 60,78,706,  literacy rate was increased by 12.81% and area is increased by 2 sq.km.
### 22. West bengal population was increased by 1,10,54,944,  literacy rate was increased by 7.66% and area is same.
### 23. Assam population was increased by 45,67,169,  literacy rate was increased by 8.89% and area is same.
### 24. Karnataka population was increased by 83,61,339,  literacy rate was increased by 8.76% and area is same.
### 25. Andhra pradesh population was increased by 88,53,236,  literacy rate was increased by 6.52% and area is same.
### 26. Maharashtra population was increased by 1,56,22,086,  literacy rate was increased by 5.44% and area is same.
### 27. Dadar & Nagar Haveli population was increased by 1,23,258,  literacy rate was increased by 18.64% and area is same.
### 28. Daman & Diu population was increased by 85,188,  literacy rate was increased by 8.9% and area is decreased by 1 sq.km.
### 29. Gujrat population was increased by 98,42,700,  literacy rate was increased by 5.62% and area is increased by 220 sq.km.
### 30. Andaman & Nicobar island population was increased by 24,316,  literacy rate was increased by 5.33% and area is same.
### 31. Pondicherry population was increased by 2,74,124,  literacy rate was increased by 4.65% and area is increased by 11 sq.km.
### 32. Tamilnadu population was increased by 1,00,36,191  literacy rate was increased by 6.59% and area is increased by 2 sq.km.
### 33. Kerala population was increased by 15,67,442,  literacy rate was increased by 3.1% and area is decreased by 11 sq.km.
### 34. Lakshadweep population was increased by 3,878,  literacy rate was increased by 5.15% and area is decreased by 2 sq.km.
### 35. Goa population was increased by 1,14,547,  literacy rate was increased by 6.7% and area is same.
##  Overall we can say  that in between 2001 and 2011 the Indian population increased to 1.21 billion with a decadal growth of 17.70%. Adult literacy rate increased to 74.04% with a decadal growth of 9.21%.



# Analysis on observation:
### India has world's geographical area 2.4% and world's population percentage 17.64.
### Lowest populatin growth rate in Lakshadweep.
### Highest populatin growth rate in Dadar & Nagar Haveli.
### Highest Urban population in Maharashtra
### Highest population state is Uttar pradesh
### Lowest population state is Sikkim
### Top 4 states in population Uttar pradesh, Maharashtra, Bihar, West Bengal
### Highest literacy rate increase in Dadar & Nagar Haveli in UT and in states in Bihar
### Highest literacy rate state is Keral
### Lowest literacy rate state is Bihar
### Top 5 states in literacy rate Keral, Lakshdweep, Mizoram, Goa, Tripura
### Bottom 5 states in literacy rate Bihar, Arunachal pradesh, Rajasthan, Jharkhand, Andhra pradesh
### Highest area increase between 2010 and 2011 in Gujrat.
### Lowest area increase between 2010 and 2011 in Chattisgarh
### Highest area decrease between 2010 and 2011 in Keral
### Lowest area decrease between 2010 and 2011 in Daman & Diu
### Most of the states has same area
### India's Total population is 1,21,08,54,977, total number of male 62,32,70,258, total population of female 58,75,84,719 and density is 382 per sq km according to 2011 census data.



